### PR TITLE
Fix completion of keyword selectors

### PR DESCRIPTION
**Pull request by guillep**
- Extend rubric iterate words backward and forward by selector when in smalltalk mode.
- Added completion tests for #10437

As a nice side effect, navigating code with ctrl+arrow now takes into account colons in keyword selectors too!

### CODE REVIEW

**Review by dionisiydk — APPROVED** [review]
That's nice. 
It would be cool to also implement the navigation between words inside selector. It was a feature in some first versions of Pharo or Squeak.

File: src/Rubric/RubSmalltalkEditor.class.st [hunk 1]
```diff
@@ -818,6 +818,12 @@ RubSmalltalkEditor >> isSmalltalkEditor [
 	^ true 
 ]
 
+{ #category : #private }
+RubSmalltalkEditor >> isWordCharacter: aCharacter [
+	"In smalltalk code, colons make part of selectors too and are part of a word"
+	^ aCharacter isAlphaNumeric or: [aCharacter = $:]
```
Comment (kasperosterbye) [inline src/Rubric/RubSmalltalkEditor.class.st]: I am able to get into trouble on this if I try to edit an assignment where there is no space between the variable and the `:=`. 
image
when I start typing `dumb` I get a completion for `dumb`, accepting the completion removes the `:`

Comment (guillep) [reply src/Rubric/RubSmalltalkEditor.class.st]: Very good catch! Thanks Kasper, I'll make a test on that.

File: src/Rubric/RubSmalltalkEditor.class.st [hunk 2]
```diff
@@ -819,9 +819,18 @@ RubSmalltalkEditor >> isSmalltalkEditor [
...
+	"By default, group alphanumeric and non-alphanumeric separately"
+	| character |
+	character := aString at: index.
 	"In smalltalk code, colons make part of selectors too and are part of a word"
-	^ aCharacter isAlphaNumeric or: [aCharacter = $:]
+	^ character isAlphaNumeric or: [
+		"Check that it is a keyword selector.
+		 - followed by a colon
+		 - but not by an assignment "
+		character = $: and: [
```
Comment (kasperosterbye) [inline src/Rubric/RubSmalltalkEditor.class.st:831]: Looks good now. I do not believe there are any other loop-holes.

**Comment by kasperosterbye** [comment]
I do not know if this is the same problem, or a separate issue. But let me try to explain.

If I start writing a new method which I want to be named say `asCamel`. When I finish writing the method name and press return to get to the next line, the autocompleter expands my method to be `asCamelCase`. Then the only way to compete the name `asCamel` is to press escape to stop the proposal, and then return. 

Perhaps I am just unfit for automatic popup :(

**Pull request merged**